receiver/apachedruidreceiver/metrics.go

```diff
@@ -0,0 +1,306 @@
+		}
+	}()
+
+	return nil
+}
+
+func (mReceiver *metricsReceiver) handleRequest(rw http.ResponseWriter, request *http.Request) {
+	if mReceiver.config.BasicAuth != nil {
+		auth := request.Header.Get("Authorization")
+		credentials := fmt.Sprintf("%s:%s", mReceiver.config.BasicAuth.Username, mReceiver.config.BasicAuth.Password)
+		configAuth := fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(credentials)))
+		if auth == "" {
+			rw.WriteHeader(http.StatusUnauthorized)
+			mReceiver.logger.Debug("Got request with no basic auth credentials when they were specified in config, dropping...")
+			return
+		} else if auth != configAuth {
+			rw.WriteHeader(http.StatusUnauthorized)
+			mReceiver.logger.Debug("Got request with incorrect basic auth credentials when they were specified in config, dropping...")
+			return
+		}
+	}
+
+	if request.Method != "POST" {
+		rw.WriteHeader(http.StatusBadRequest)
+		mReceiver.logger.Debug("Receiver server only accepts POST requests", zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	if request.Header.Get("Content-Type") != "application/json" {
+		rw.WriteHeader(http.StatusBadRequest)
+		errMessage := "Content type must be JSON"
+		if _, err := rw.Write([]byte(errMessage)); err != nil {
+			mReceiver.logger.Debug("Error writing response message to Druid", zap.Error(err), zap.String("remote", request.RemoteAddr))
+			return
+		}
+		mReceiver.logger.Debug(errMessage, zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	payload, err := io.ReadAll(request.Body)
+	if err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		mReceiver.logger.Debug("Failed to read metrics payload", zap.Error(err), zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	var metrics []datapoint
+	if err = json.Unmarshal(payload, &metrics); err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		mReceiver.logger.Debug("Failed to convert metrics payload from JSON array to golang slice", zap.Error(err), zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	purgedMetrics, service := purgeMetrics(metrics)
+
+	if len(purgedMetrics) == 0 {
+		rw.WriteHeader(http.StatusOK)
+		mReceiver.logger.Debug("No relevant metrics provided by request", zap.String("remote", request.RemoteAddr))
+		return
+	}
+
+	if err := mReceiver.consumer.ConsumeMetrics(request.Context(), mReceiver.processMetrics(purgedMetrics, service)); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		mReceiver.logger.Error("Failed to consume payload as metric", zap.Error(err))
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+
+func (mReceiver *metricsReceiver) processMetrics(metrics []datapoint, service string) pmetric.Metrics {
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	queryCountStats := make(map[string]float64)
+	for _, dataPoint := range metrics {
+		queryCountStats[dataPoint.Metric] = dataPoint.Value
+	}
+
+	mReceiver.metricsBuilder.RecordApachedruidQueryCountDataPoint(now, int64(queryCountStats[druidQueryCount]))
```

Comment:
This assumes each metrics is present, should check that the metric was provided in payload before recording.